File: 4/python/main.py
def reverse_string(produto) -> str :
    string = str(produto)
    newString = ""
    for i in range(len(string)):
        newString += string[len(string)-i-1]
    return newString

def check_palindrom(produto) -> bool :
    return str(produto) == reverse_string(produto)

def helper(produto, largest) -> int :
    if not check_palindrom(produto) :
        return largest
    if produto >= largest :
        return produto
    return largest

def solve() -> int :
    largest = 0
    for i in range(100, 1000):
        for k in range(100, 1000):
            produto = i * k
            largest = helper(produto, largest)

    return largest

def main() -> None :
    print(solve())


if __name__ == "__main__" :
          main()

File: 4/js/main.js
function reverseString(produto) {
	var pString = produto.toString();
	var reversedString = "";
		for (var i = 0; i < pString.length; i++) {
		reversedString += pString[pString.length-i-1];	
	}
	return reversedString;
}

function checkPalindrom(produto) {
	return produto.toString() == reverseString(produto);
}

function helper(p, l) {
	if (!checkPalindrom(p)){
		return l;
	}
	if (p >= l) {
		return p;
	}
	return l;
}

function solve(){
	var l = 0;
	for (var i = 0; i < 1000; i++){
		for (var k = 0; k < 1000; k++){
			var p = k * i;
			l = helper(p, l);
		}
	}
	return l;
}

console.log(solve());
